### PR TITLE
Handle scalable store size in MemCpyOptimizer

### PR DESCRIPTION
**Pull request by momchil-velikov**
The compiler crashes with an ICE when it tries to create a `memset` with scalable size.

### CODE REVIEW

File: llvm/test/CodeGen/AArch64/memset-scalable-size.ll
Comment (nikic) [inline llvm/test/CodeGen/AArch64/memset-scalable-size.ll:1]: Should be in llvm/test/Transforms/MemCpyOpt.

**Review by jthackray — APPROVED** [review]
Thanks for the fix. LGTM.

**Review by nikic — COMMENTED** [review]
Is it actually desirable to convert scalable load/store into a memset?

**Review by nikic — CHANGES_REQUESTED** [review]
.

File: llvm/test/CodeGen/AArch64/memset-scalable-size.ll
Comment (momchil-velikov) [reply llvm/test/CodeGen/AArch64/memset-scalable-size.ll:1]: Done

**Review by nikic — APPROVED** [review]
LGTM

**Pull request merged**